### PR TITLE
Bug 1217046 - Move FirefoxTestCase from firefox_ui_harness into firefox_puppeteer

### PR DESCRIPTION
**Pull request by ujjwalwahi**

### CODE REVIEW

File: firefox_puppeteer/testcase.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
```
Comment (whimboo) [inline firefox_puppeteer/testcase.py]: Great. But I would really suggest to keep the subfolder and put this file into firefox_puppeteer/testcases/. Maybe we can keep using `base.py`?

File: firefox_puppeteer/testcase.py [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 from marionette import MarionetteTestCase
 
-from firefox_puppeteer import Puppeteer
-from firefox_puppeteer.ui.browser.window import BrowserWindow
+from . import Puppeteer
```
Comment (whimboo) [inline firefox_puppeteer/testcase.py]: Please keep the absolute references for imports.

File: firefox_ui_harness/runners/base.py [hunk 3]
```diff
@@ -7,8 +7,8 @@
 import mozinfo
 from marionette import BaseMarionetteTestRunner
 
-from firefox_ui_harness.testcases import FirefoxTestCase
 import firefox_ui_tests
+from firefox_puppeteer.testcases import FirefoxTestCase
```
Comment (whimboo) [inline firefox_ui_harness/runners/base.py:11]: In a follow-up bug we will have to get rid of this import given that the harness should not reference puppeteer. For now this looks fine given that we at least have extracted the testcases from the harness.

File: firefox_ui_tests/testcases/update.py [hunk 4]
```diff
@@ -4,9 +4,10 @@
 
 import pprint
 
+from firefox_puppeteer.testcases import FirefoxTestCase
```
Comment (whimboo) [inline firefox_ui_tests/testcases/update.py]: Please sort this line alphabetically into the next block.

**Comment by whimboo** [comment]
Just a nit to fix. If you want you can already squash all the commits into a single one and give it a proper comment (see git log for examples), so it's easier for landing later today. Thanks!

**Comment by whimboo** [comment]
@ujjwalwahi if you do changes to your pull I would recommend that you add a comment to the appropriate bug in Bugzilla or at least here on the PR. Sadly Github doesn't inform anyone if new commits have been pushed. So your latest commits went unnoticed. Sorry.

**Comment by ujjwalwahi** [comment]
@whimboo actually I don't know why but one test is failing during Travis CI build, I was trying  to fix that, that's why I didn't added a comment. Yet, I am not unable to find a solution. It will be great if you can suggest  something. Code seems okay to me.

**Comment by whimboo** [comment]
I see. So the failure here is only happening for a single Travis build.  All three others are passing. By checking the failure I see that this is [bug 1212811](https://bugzilla.mozilla.org/show_bug.cgi?id=1212811) and not related to this PR at all. So all fine here. I will have a look at the PR tomorrow morning.

**Comment by whimboo** [comment]
Looks all great to me now! Even the commit message is perfect so that the PR can directly be merged. I will do it right away.

**Comment by whimboo** [comment]
Merged the PR as https://github.com/mozilla/firefox-ui-tests/commit/50527c612f3c063ab52046d0d01f610b64f5f003

**Pull request closed without merging**